test/jdk/javax/print/StreamPrintingOrientation.java

```diff
@@ -24,95 +24,110 @@
 /*
  * @test
  * @bug 4904236
- * @summary You would see a cross-platform print dialog being popped up. Check whether orientation is shown as LANDSCAPE. Click 'OK'. 'streamexample.ps' will be created in the same dir where this application was executed. Pass if the orientation in the ps file is landscape.
- * @run main/manual StreamPrintingOrientation
+ * @key printer
+ * @summary StreamPrintService ignores the PrintReqAttrSet when printing through 2D Printing
+ * @run main StreamPrintingOrientation
  */
 
-import java.awt.*;
-import java.awt.print.*;
-import javax.print.*;
-import javax.print.attribute.standard.*;
-import javax.print.attribute.*;
-import java.io.FileOutputStream;
 import java.io.File;
-import java.util.Locale;
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import javax.print.attribute.Attribute;
+import javax.print.PrintService;
+import javax.print.StreamPrintServiceFactory;
+import javax.print.attribute.standard.Copies;
+import javax.print.attribute.standard.JobName;
+import javax.print.attribute.standard.OrientationRequested;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+
+public class StreamPrintingOrientation implements Printable {
+
+    public static void main(String[] args) throws Exception {
+        StreamPrintingOrientation pd = new StreamPrintingOrientation();
+        PrinterJob pj = PrinterJob.getPrinterJob();
+        HashPrintRequestAttributeSet prSet = new HashPrintRequestAttributeSet();
+        PrintService service = null;
+
+        FileOutputStream fos = null;
+        File fl = null, fp = null;
+        String mType = "application/postscript";
 
-class StreamPrintingOrientation implements Printable {
-        /**
-         * Constructor
-         */
-         public StreamPrintingOrientation() {
-                super();
+        fl = new File("stream_landscape.ps");
+        fl.deleteOnExit();
+        fos = new FileOutputStream(fl);
```

Comment:
Can be rearranged something like..

    String mType = "application/postscript";
    File fp = null;
    File fl = new File("stream_landscape.ps");
    fl.deleteOnExit();
    FileOutputStream fos = new FileOutputStream(fl);